codegenerator/cli/templates/static/codegen/src/InMemoryStore.res

```diff
@@ -0,0 +1,94 @@
+@genType
+type rawEventsKey = {
+  chainId: int,
+  eventId: string,
+}
+
+let hashRawEventsKey = (key: rawEventsKey) =>
+  EventUtils.getEventIdKeyString(~chainId=key.chainId, ~eventId=key.eventId)
+
+@genType
+type dynamicContractRegistryKey = {
+  chainId: int,
+  contractAddress: Address.t,
+}
+
+let hashDynamicContractRegistryKey = ({chainId, contractAddress}) =>
+  EventUtils.getContractAddressKeyString(~chainId, ~contractAddress)
+
+module EntityTables = {
+  type t = dict<InMemoryTable.Entity.t<Entities.internalEntity>>
+  exception UndefinedEntity(string)
+  let make = (entities: array<module(Entities.InternalEntity)>): t => {
+    let init = Js.Dict.empty()
+    entities->Belt.Array.forEach(entity => {
+      let module(Entity) = entity
+      init->Js.Dict.set(Entity.key, InMemoryTable.Entity.make())
+    })
+    init
+  }
+
+  let get = (type entity, self: t, entityMod: module(Entities.Entity with type t = entity)) => {
+    let module(Entity) = entityMod
+    switch self->Utils.Dict.dangerouslyGetNonOption(Entity.key) {
+    | Some(table) =>
+      table->(
+        Utils.magic: InMemoryTable.Entity.t<Entities.internalEntity> => InMemoryTable.Entity.t<
+          entity,
+        >
+      )
+
+    | None =>
+      UndefinedEntity(Entity.key)->ErrorHandling.mkLogAndRaise(
+        ~msg="Unexpected, entity InMemoryTable is undefined",
+      )
+    }
+  }
```

Comment:
I've refactored inMemoryStore to be library like where it does not codegen a record for each entity but uses a dict to hold each entity.

If you look at the commits you should be able to see the refactor diff before the file move.